src/QuadForm/Lattices.jl

```diff
@@ -1181,34 +1181,49 @@ end
 
 function maximal_sublattices(L::AbsLat, p; use_auto::Bool = false,
                                            callback = false, max = inf)
-  @req base_ring(L) == order(p) "asdsd"
+  @req base_ring(L) == order(p) "Incompatible arguments: p must be an ideal in the base ring of L"
 
   B = local_basis_matrix(L, p, type = :submodule)
   n = nrows(B)
   R = base_ring(L)
   K = nf(R)
   k, h = ResidueField(R, p)
   hext = extend(h, K)
+  use_auto = isdefinite(L) ? use_auto : false
 
   if use_auto
-    throw(NotImplemented())
+    G = automorphism_group_generators(L)
+    Binv = inv(B)
```

Comment:
are you sure that `B` is always square? For instance if the lattice is not full.
perhaps you rather need `can_solve_with_solution`

Reply:
This is what I wanted to do at first, but the outputs were wrong. So I chose to follow something already implemented for neighbours methods. I could try to manage it and see how to adapt it properly here to deal with the non-full cases

Reply:
If it is too much of a mess, set it as a `@req` and add a todo.

Reply:
I guess we should not allow the use of the automorphism group for non full lattices since it cannot compute the generators of the automorphism group

Reply:
No sorry I said nothing, we can make it by disallowing the `ambient_representation`.